recipes/enjincppsdk/all/conanfile.py

```diff
@@ -0,0 +1,133 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.43.0"
+
+
+class EnjinCppSdk(ConanFile):
+    name = "enjincppsdk"
+    description = "A C++ SDK for development on the Enjin blockchain platform."
+    license = "Apache-2.0"
+    topics = ("enjin", "sdk", "blockchain")
+    homepage = "https://github.com/enjin/enjin-cpp-sdk"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    settings = "os", "compiler", "arch", "build_type"
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_default_http_client": [True, False],
+        "with_default_ws_client": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": False,
+        "with_default_http_client": False,
+        "with_default_ws_client": False,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "16",
+            "gcc": "9",
+            "clang": "9",
+        }
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        if self.options.with_default_http_client:
+            self.options["cpp-httplib"].with_openssl = True
+
+        self.options["spdlog"].header_only = True
+
+    def requirements(self):
+        if self.options.with_default_http_client:
+            self.requires("cpp-httplib/0.8.5")
+
+        if self.options.with_default_ws_client:
+            self.requires("ixwebsocket/11.0.4")
+
+        self.requires("rapidjson/1.1.0")
+        self.requires("spdlog/1.8.2")
+
+    def validate(self):
+        # Validations for OS
```

Comment:
```suggestion
        # Validations for OS
        if self.settings.compiler == "clang" and self.settings.compiler.libcxx in ["libstdc++", "libstdc++11"] and self.settings.compiler.version == "11":
            raise ConanInvalidConfiguration("clang 11 with libstdc++ is not supported due to old libstdc++ missing C++17 support")
            
```
see https://github.com/conan-io/conan-center-index/pull/9063#issuecomment-1021175018